R/detect.R

```diff
@@ -119,30 +119,69 @@ str_ends <- function(string, pattern, negate = FALSE) {
 #' * `_` matches a single character (like `.`).
 #' * `%` matches any number of characters (like `.*`).
 #' * `\%` and `\_` match literal `%` and `_`.
-#' * The match is case insensitive by default.
+#' * The match is case sensitive.
 #'
 #' @inheritParams str_detect
 #' @param pattern A character vector containing a SQL "like" pattern.
 #'   See above for details.
-#' @param ignore_case Ignore case of matches? Defaults to `TRUE` to match
-#'   the SQL `LIKE` operator.
+#' @param ignore_case `r lifecycle::badge("deprecated")`
 #' @return A logical vector the same length as `string`.
 #' @export
 #' @examples
 #' fruit <- c("apple", "banana", "pear", "pineapple")
 #' str_like(fruit, "app")
 #' str_like(fruit, "app%")
+#' str_like(fruit, "APP%")
 #' str_like(fruit, "ba_ana")
-#' str_like(fruit, "%APPLE")
-str_like <- function(string, pattern, ignore_case = TRUE) {
+#' str_like(fruit, "%apple")
+str_like <- function(string, pattern, ignore_case = deprecated()) {
+  check_lengths(string, pattern)
+  check_character(pattern)
+  if (inherits(pattern, "stringr_pattern")) {
+    cli::cli_abort("{.arg pattern} must be a plain string, not a stringr modifier.")
+  }
+  if (lifecycle::is_present(ignore_case)) {
+    lifecycle::deprecate_warn(when = "1.5.2",
+                              what = "str_like(ignore_case)",
+                              details = "str_like() is always case sensitive. Use
+                              str_ilike() for case insensitive string matching.")
+  }
+
+  pattern <- regex(like_to_regex(pattern), ignore_case = FALSE)
+  stri_detect_regex(string, pattern, opts_regex = opts(pattern))
+}
+
+#' Detect a pattern in the same way as `SQL`'s `ILIKE` operator
```

Comment:
I think you could combine the docs for these two functions. It'd also be useful to add a note about the incorrect behaviour of older versions of `str_like()`

Reply:
@hadley by this did you mean to use @inheritParams str_like? If so, I've added this here
https://github.com/tidyverse/stringr/pull/544/commits/1a366e3495422e1fec13495a2cd4bea58774baca But please let me know if I've misunderstood as I wasn't sure of what you meant

Reply:
I meant using `@rdname`; see the changes I just pushed.